CONTRIBUTING.md

```diff
@@ -1,51 +1,150 @@
+    repositories to keep track of: the original ManimCommunity repo, your own
+    fork of it, and your local repository.
 
-3. Now that you have cloned the repo, before making any changes, you have to enter the commands below in order to be able to contribute to Manim Community.
+3. You have to make `git` be aware of the three repositories, by entering the
+   commands below.
 
    ```sh
-   git remote add fork https://github.com/ManimCommunity/manim.git
-   git fetch fork
-   git checkout -b <your-branch-name> fork/master
+   git remote add upstream https://github.com/ManimCommunity/manim.git
+   git pull upstream master
    ```
 
-   The first command tells `git` that you are going to make a Pull Request to Manim Community. 
-   The second command pulls all the commits from the aforementioned fork.
-   Finally, the third one makes your current working branch up-to-date with Manim Community's master branch.
-
-    Now there are three repositories that git is keeping track of: the manim community repo (referred to as "fork"), your own fork of it (referred to by git as "origin"), and your local repository.
+	After these commands, your local repository can keep track of your fork
+    (referred to as 'origin') as well as the main ManimCommunity repository
+    (referred to as 'upstream').
 
-4. After that, you can make your changes to the repo's files (the code is in the `manim` directory). Then, you can commit said changes.
+4. You can now make any changes to your local repository, add them, and commit
+   them.
 
-> As stated in the General Contribution Guidelines, make sure to test your changes against the example scenes in the `example_scenes` directory and test scenes in the `tests` directory, and add a new test in the `tests` directory for your new feature if you've made one.
-
-5. Finally, instead of  typing in `git push`, enter the command below.
+5. To make your changes visible in your GitHub fork, instead of typing in `git
+   push` as usual, you need to enter the command below.
 
    ```sh
    git push -u origin <your-branch-name>
    ```
 
-   Doing so creates a new branch with the updated contents of your fork on GitHub.
+   Doing so creates a new branch with the updated contents of your fork on
+   GitHub (the 'origin').
+
+6. If you want your changes to be incorporated to the main ManimCommunity
+   repository, you need to make sure that there are no merge conflicts between
+   the current upstream/master and the changes you are trying to make.  For
+   this purpose, please execute `git pull origin master` before you push
+   changes to your fork.  If this generates merge conflicts, make sure to
+   resolve them before submitting a PR.
+
+7. Further, make sure that you run the tests locally and that they all pass
+   before submitting a PR.  You can run the tests locally by executing `pytest`
+   from the main `manim/` folder (do not execute pytest from the code folder
+   `manim/manim`).
```

Comment:
Is this really necessary? 
I normally don't do this.
I thought that github tests are there for this exact reason, that you don't have to run them locally.
Especially for new contributers, I think it would be nice to keep it as simple as possible

Reply:
Hmmm I disagree here. Asking people to test locally will avoid 300 commits trying to fix tests on GitHub action.

Reply:
> Is this really necessary? I normally don't do this.

You should!

> I thought that github tests are there for this exact reason, that you don't have to run them locally.

No. github tests run the tests on many platforms and many versions of different things. The author of a PR should be expected to run tests locally.

> Especially for new contributers, I think it would be nice to keep it as simple as possible

I assume you mean "inexperienced contributors" and not "new contributors". A new contributor to manim may already have a lot of experienced with open source and not be intimidate by local testing. Further, I don't think that asking people to run a single command with a single word (`pytest`) is asking too much.